### DISCUSSION TITLE
Deploying nextjs onto AWS EC2 with PM2

### DISCUSSION THREAD

**Discussion by crslegend** [vercel/next.js#19728] [Help] [answered]
hi all,

I have set up a nextjs project using `npx create-next-app` and I wish to test out the deployment process with the basic skeleton codes from `npx create-next-app`. I have created an EC2 instance running on Ubuntu v18

I tried running the nextjs using pm2 with the following command:
`npm run build
pm2 start npm --name "next" -- prod`

However, I'm getting `App [nextjs:0] exited with code [1] via signal [SIGINT]` upon execution of the command and nothing is appearing on my EC2 public IP address.

Below is the snippet from the `package.json` of the nextjs project:
`"scripts": {
    "dev": "next dev",
    "build": "next build",
    "start": "next start",
    "prod": "next start -p 80",
    "export": "next export"
  },`

Does anyone have any suggestions on how I can deploy in the correct manner?

**Comment by crslegend** [comment, marked as answer]
I found a solution:

`sudo pm2 start npm --name "<name>" -- run prod`

**Comment by jamesmosier** [comment]
Glad you got it working! This is what I use for pm2 on EC2/EB instances:

`npm install pm2`

```js
"start": "next start -p $PORT",
"start-pm2": "./node_modules/pm2/bin/pm2-runtime start npm --name 'my-app-name' -- start",
```

and then I change the startup script to be `npm run start-pm2` for my Elastic Beanstalk instance,

**Reply by manu-sparheld** [reply]
@jamesmosier  Did you try it with monorepos which have different workspaces in next application? Thanks
